### DISCUSSION TITLE
Calling overloaded function with same params length

### DISCUSSION THREAD

**Discussion by fdarian** [wevm/viem#1799] [Question]
Hi team,

I'm currently creating a ViemAdapter for [SafeCoreSdk](https://github.com/safe-global/safe-core-sdk/pull/665). I've come across a function that is overloaded with the same number of parameters but with different types.

For instance, consider the [CompatiblityFallbackHandler:isValidSignature](https://github.com/safe-global/safe-core-sdk/blob/6bb692d64ab0bba4c1d40706e1458ad0ab0a6d2b/packages/protocol-kit/src/Safe.ts#L1461), which includes `isValidSignature(bytes32,bytes)` and `isValidSignature(bytes,bytes)`.

In such cases, how can I specify which version of the function to use?

For reference
```ts
const abi = [
    {
      inputs: [
        { internalType: 'bytes32', name: '_dataHash', type: 'bytes32' },
        { internalType: 'bytes', name: '_signature', type: 'bytes' }
      ],
      name: 'isValidSignature',
      outputs: [ { internalType: 'bytes4', name: '', type: 'bytes4' } ],
      stateMutability: 'view',
      type: 'function'
    },
    {
      inputs: [
        { internalType: 'bytes', name: '_data', type: 'bytes' },
        { internalType: 'bytes', name: '_signature', type: 'bytes' }
      ],
      name: 'isValidSignature',
      outputs: [ { internalType: 'bytes4', name: '', type: 'bytes4' } ],
      stateMutability: 'view',
      type: 'function'
    }
] as const
```

**Comment by d4mr** [comment]
This is currently not possible. A workaround is to split the ABIs and use each separately to specify which overloaded method to use.

(Interaction with @tmm on x: https://twitter.com/awkweb/status/1759788457491145192?t=APAXKJKK3jSAG6gVJyIL6Q&s=19)

**Reply by tmm** [reply]
To be clear, overloads do generally work. This just could be an edge case. Haven't tested it so not sure. Feel free to add a breaking test as a PR and I can take a look.
